pkg/services/sqlstore/alert.go

```diff
@@ -151,13 +151,20 @@ func HandleAlertsQuery(query *m.GetAlertsQuery) error {
 
 func deleteAlertDefinition(dashboardId int64, sess *DBSession) error {
 	alerts := make([]*m.Alert, 0)
-	sess.Where("dashboard_id = ?", dashboardId).Find(&alerts)
+	if err := sess.Where("dashboard_id = ?", dashboardId).Find(&alerts); err != nil {
+		return err
+	}
 
+	var gotErr error
 	for _, alert := range alerts {
-		deleteAlertByIdInternal(alert.Id, "Dashboard deleted", sess)
+		if err := deleteAlertByIdInternal(alert.Id, "Dashboard deleted", sess); err != nil {
+			if gotErr == nil {
```

Comment:
Is `gotErr` going to be something other than nil?

Reply:
@papagian Yes, if one of the calls to `deleteAlertByIdInternal` during the loop fails, `gotErr` will be set to the error in question.

Reply:
I see, in that case, since the first encountered error is returned, i think it would be preferable to return directly this error instead of waiting for the loop to finish.

Reply:
@papagian I could actually use some input on this, but I thought it would be best to clean up as many of the alerts as possible before failing?

Reply:
That totally makes sense. The best approach then I think would be to return all the errors instead of the first one but I'm not sure if this worths the extra effort. @marefr what  do you think?

Reply:
I'm not sure how one would combine all the errors myself.

Reply:
Seems like all code that calls `deleteAlertByIdInternal` runs in a transaction so in that case I would suggest failing fast since everything will be rolled back on error.

Reply:
good point!

Reply:
@marefr OK, I'll make the change along with a comment to clarify the reason.